gitlab/v4/objects/groups.py

```diff
@@ -177,18 +194,24 @@ def share(self, group_id, group_access, expires_at=None, **kwargs):
         Raises:
             GitlabAuthenticationError: If authentication is not correct
             GitlabCreateError: If the server failed to perform the request
+
+        Returns:
+            Group
         """
         path = f"/groups/{self.get_id()}/share"
         data = {
             "group_id": group_id,
             "group_access": group_access,
             "expires_at": expires_at,
         }
-        self.manager.gitlab.http_post(path, post_data=data, **kwargs)
+        server_data = self.manager.gitlab.http_post(path, post_data=data, **kwargs)
+        if TYPE_CHECKING:
+            assert isinstance(server_data, dict)
+        return Group(self.manager, server_data)
```

Comment:
@nejch This I'm not 100% sure about.

Maybe it should be this instead:

```
manager = GroupManager(self.manager.gitlab, parent=self.manager._parent)
return Group(manager, server_data)
```

Reply:
Since we already have the group, should this be enough?

```suggestion
        self._update_attrs(server_data)
```
The API docs are not super clear but it should be the same group details that are returned, not the other group I hope - that would be confusing :grin:

Reply:
Hmm. I'm not sure... 🤷‍♂️

I'm not sure which group details it returns. The group we are sharing or the group we shared it with?

Reply:
I just checked with
```python
gl.http_post('/groups/123/share', group_id=456, group_access=30)
```

And the json response had the details from the group we are sharing :) So this should be ok.

Reply:
Cool and thanks! Updated it with your suggestion